source/Calamari/Deployment/PackageRetention/Repositories/JsonJournalRepositoryFactory.cs

```diff
@@ -1,23 +1,31 @@
 ﻿using System;
+using System.IO;
 using Calamari.Common.Features.Processes.Semaphores;
 using Calamari.Common.Plumbing.FileSystem;
+using Calamari.Common.Plumbing.Variables;
 
 namespace Calamari.Deployment.PackageRetention.Repositories
 {
     public class JsonJournalRepositoryFactory : IJournalRepositoryFactory
     {
-        internal const string DefaultJournalName = "PackageRetentionJournal.json";
+        const string DefaultJournalName = "PackageRetentionJournal.json";
 
         readonly ICalamariFileSystem fileSystem;
         readonly ISemaphoreFactory semaphoreFactory;
         readonly string journalPath;
 
-        public JsonJournalRepositoryFactory(ICalamariFileSystem fileSystem, ISemaphoreFactory semaphoreFactory)
+        public JsonJournalRepositoryFactory(ICalamariFileSystem fileSystem, ISemaphoreFactory semaphoreFactory, IVariables variables)
         {
             this.fileSystem = fileSystem;
             this.semaphoreFactory = semaphoreFactory;
 
-            this.journalPath = @"C:\Octopus\PackageJournal.json";//journalPath;
+            var packageRetentionJournalPath = variables.Get(KnownVariables.Calamari.PackageRetentionJournalPath);
+            if (packageRetentionJournalPath == null)
+            {
+                var tentacleHome = variables.Get(TentacleVariables.Agent.TentacleHome) ?? string.Empty; // Retention is only enabled when TentacleHome exists
+                packageRetentionJournalPath = Path.Combine(tentacleHome, DefaultJournalName);
```

Comment:
`Path.Combine` requires a non-null string so if we don't set `tentacleHome` to something here (at the moment I went with an empty string) then it would fail in the integration tests again.